libs/atqr/domain/src/lib/domain-spec/services/User.disl

```diff
@@ -0,0 +1,49 @@
+domain ATQR
+
+  context User
+    repository UserRepository
+
+      createUser: Contact
+        rules:
+          - Contact.email does not exist in UserRepository.users.email
+        create User and return
+
+      findUserById :userId
+        ⮐ User or null
+
+      findUserByEmail :email
+        ⮐ User or null
+
+      updateUser :user
+        rules:
+          - User.email does not exist in UserRepository.users.email
+
+        update User and return
+
+
+    service UserService
+      depends on UserRepository
+
+      -> CreateUser? :contact
+          either createUser :contact using UserRepository
+              or show UserAlreadyExists screen
+
+      -> GetUser? :userId
+          either findUserById :usereId using UserRepository
+              or show UserNotFound screen
+
+      -> GetUserByEmail? :email
+          either findUserByEmail :email using UserRepository
+              or show UserNotFound screen
+
+      -> UserCreateChallenge? :userId
+          either challenge :user using UserRepository
+              or show UserNotFound screen
+
+      -> UserPlayChallenge? :userId
+          either up Enrollment :user using UserRepository
+              or show UserNotFound screen
+
+      -> UserOfficiateChallenge? :userId
+          either officiate :user using UserRepository
+              or show UserNotFound screen
```

Comment:
These don't make sense. I mean the phrases in English. Don't be afraid of writing long phrases in the spec. 

E.g.: [43] "UserPlayChallenge". I don't know if you mean "AddUserAsChallengePlayer" or "DoesTheUserPlayAnyChallenges?". 

Btw, you are marking methods as if they were commands (adding `?` suffix). These are not actions directly taken by the user thus cannot be commands 

Other issues:

- 44: What does "up" Enrollment mean?
- 47: Same issues as in line 43
- 48: you can't "officiate" a user and if you are using the UserRepository for that there should exist an `officiate` method in it